include/flatbuffers/code_generators.h

```diff
@@ -74,9 +74,35 @@ class BaseGenerator {
     return true;
   }
 
-  std::string FullNamespace(const char *separator, const Namespace &ns) {
+  static void NamespacePrefixAsComponents(const std::string &prefix, std::vector<std::string> &namespaces)
+  {
+    std::string::size_type componentStart = 0;
+    std::string::size_type componentEnd = prefix.find_first_of('.', componentStart);
+    while (componentEnd != std::string::npos) {
+      namespaces.push_back(prefix.substr(componentStart, componentEnd - componentStart));
+      componentStart = componentEnd + 1;
+      componentEnd = prefix.find_first_of('.', componentStart);
```

Comment:
use `for (;;) { .. if (..) break; .. }` instead of `while` to avoid duplicating code like searching for `.`